src/logworker.cpp

```diff
@@ -120,6 +120,20 @@ namespace g3 {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, sink] {
+         for (auto it = _impl._sinks.begin(); it != _impl._sinks.end(); ) {
+            if ((*it).get() == sink.get()) {
+               it = _impl._sinks.erase(it);
+            } else {
+               ++it;
+            }
+         }
+      };
+      auto token_done = g3::spawn_task(bg_removesink_call, _impl._bg.get());
```

Comment:
I am sure what bothers you about nullptr. I am locking weak_ptr which may result to null but then it will be safely handled by shared_ptr == operator comparison (at least that was the idea).
I will provide unit tests for remove sink functionality ;)